### ISSUE TITLE
Help with control

### ISSUE THREAD

**Issue by blusius** [JohMah/Hoverboard-JC2015.7.31.V2.0-Hack#3]
Hi guys I have trouble to control motors. 
I built and flashed using Keil successfully, even tried to debug. 
Seems even hardcoded values SetEnable(SET); and SetPWM(300); does nothing.
Tried inputs on usart and external sockets, potentiometer, PWM, UART sent command packets from arduino.
Can anyone kick some info about wiring or working example pinout or something. 
Do those boards has to be connected together or can work as separate masters?

All info from other branches still not makes more sense. maybe need some soldering on the board?
@phpkadir
@johmah 

Thank you

![image](https://user-images.githubusercontent.com/103856515/163691284-b7021f1e-18ba-45d2-9da9-6dddd54b5d05.png)
![image](https://user-images.githubusercontent.com/103856515/163691302-682629c5-acc1-4110-82ea-2e988b51257d.png)
![image](https://user-images.githubusercontent.com/103856515/163691314-440b487a-b13d-4f40-9571-c07ae6ffa252.png)

**Comment by phpkadir** [comment]
[blusius](https://github.com/blusius)
[JohMah](https://github.com/JohMah)
i have prepared the software which turns the bldc motor with this awesome board.
https://github.com/phpkadir/Hoverboard-JC2015.7.31.V2.0-Hack
proof https://www.youtube.com/watch?v=XIqb7e3k_FA

**Comment by blusius** [comment]
nice man. I gave up and ordered separate controllers :) Tested your firmware. It works. Had to find correct phase wiring for original motors. YG GB BY for me spins but not with default wiring by color. Motor has some vibrations. strange but backwards goes smoother and has more speed. tested on multiple motors the same behaviour. tryed to change pwm_freq but not helps to run smoother. anyway appreciate your work.

**Comment by phpkadir** [comment]
i cant find any usable pins to attach a potentiometer to control the speed of the motor. for smooth motor run i can check it later.

**Comment by blusius** [comment]
did you find a way to control with potenciometer or pwm? that would be enough to control from raspberry.

**Comment by Ability-dis** [comment]
> [blusius](https://github.com/blusius) [JohMah](https://github.com/JohMah) i have prepared the software which turns the bldc motor with this awesome board. https://github.com/phpkadir/Hoverboard-JC2015.7.31.V2.0-Hack proof https://www.youtube.com/watch?v=XIqb7e3k_FA

Great I was looking for it. But the links are not working.
If you still working on it then please let me know.

**Comment by phpkadir** [comment]
yes its all working. the code only turns motor with hall sensors. the gyro sensor code not included by my needs. the start stop button pushes is used for speed up and speed down. u can remove the buzzer+buzzer transistor+r22 of buzzer and add a potentiometer center pin on pb1 ground and +3.3v than adjust the speed of bldc motor.

**Comment by Chuma1986** [comment]
> yes its all working. the code only turns motor with hall sensors. the gyro sensor code not included by my needs. the start stop button pushes is used for speed up and speed down. u can remove the buzzer+buzzer transistor+r22 of buzzer and add a potentiometer center pin on pb1 ground and +3.3v than adjust the speed of bldc motor.

phpkadir, can you help me. i try to use Hoverboard-JC2015.7.31.V2.0-Hack code but it doesn't work at all. can you send your working code or put it on github?

**Comment by QGB** [comment]
TIMER_BLDC_CHANNEL_G  TIMER_CCXN_ENABLE

**Comment by L1mp4** [comment]
Hi! Is there anyone who has gotten these boards to work? I dont know where to put anything and cant even flash the firmware. Appriciate if anyone can give me any pointers!
Thanks!

**Comment by QGB** [comment]
@L1mp4  GD32 must use `GD_Link_CLI.exe`  to `SetRDP 0` first

**Comment by L1mp4** [comment]
Thank you for answering. Greatly appreciated! 
Tho i am very confused of what you mean. Right now my card just beeps and shuts of, so i cant even flash. Then i wonder if you got a schematic for the headers to program? So i dont do anything wrong. The layout seams to differ? 
Thanks! 
![IMG_7767](https://github.com/JohMah/Hoverboard-JC2015.7.31.V2.0-Hack/assets/138057493/d8ef460e-69d0-4f94-943b-cc8e00968f09)

**Comment by phpkadir** [comment]
you 

> Thank you for answering. Greatly appreciated! Tho i am very confused of what you mean. Right now my card just beeps and shuts of, so i cant even flash. Then i wonder if you got a schematic for the headers to program? So i dont do anything wrong. The layout seams to differ? Thanks! ![IMG_7767](https://user-images.githubusercontent.com/138057493/252935363-d8ef460e-69d0-4f94-943b-cc8e00968f09.JPG)
your bldc driver is different than gd32 board. its original stm32f103c8t6. the difference for gd32f103c have timer0=timer1 at stm32 etc. you can find working stm32f103c bldc controller codes from casainho github [user](https://github.com/casainho)

**Comment by L1mp4** [comment]
Hi!
Thanks. I will check this link. Thank you for answering!

**Comment by L1mp4** [comment]
Thanks. Sorry. Which one would you use for my board? Also, how can i get it flashable. It just beeps, i guess since i dont have anything connected... :/

**Comment by phpkadir** [comment]
> you use for my board? Also, how can i get it flashable. It just beeps, i guess since i dont have anything connected... :/

dear friend you should find programming pins and pwm pins. i can give you modified bldc codes. i have

**Comment by zimmii91** [comment]
hi @L1mp4, did you find something working no this board? i seem to have the same one

**Comment by L1mp4** [comment]
Hi. Nah, sorry. I have no idea what to do. :-/

**Comment by zimmii91** [comment]
> Hi. Nah, sorry. I have no idea what to do. :-/

ah ok, sorry to hear that. i have 4 sets of these boards so i guess i will continue trying to figure it out.
